### PR TITLE
Update release notes master

### PR DESCRIPTION
**Pull request by orhovy**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/28875

## Description
Update-release-notes will now take new version from master, if its unreachable will take from local metadata file.
If release notes were added in the branch and a newer version exists in master, they will be renamed to the newest version.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/__main__.py [hunk 1]
```diff
@@ -899,7 +900,10 @@ def update_pack_releasenotes(**kwargs):
         sys.exit(0)
     if _packs:
         for pack in _packs:
-            if pack in packs_existing_rn and update_type is not None:
+            if pack in packs_existing_rn and update_type is None:
+                with open(packs_existing_rn[pack], 'r') as f:
```
Comment (amshamah419) [inline demisto_sdk/__main__.py:904]: Let's add some error handling here.

File: demisto_sdk/__main__.py [hunk 2]
```diff
@@ -899,7 +900,10 @@ def update_pack_releasenotes(**kwargs):
...
+                    full_text = f.read()
```
Comment (amshamah419) [inline demisto_sdk/__main__.py:905]: Consider a more intuitive variable name.

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 3]
```diff
@@ -129,6 +133,20 @@ def _does_pack_metadata_exist(self):
 
         return True
 
+    def get_master_version(self):
+        """
+        Get the current version from origin/master if available, otherwise return '0.0.0'
+        """
+        master_current_version = '0.0.0'
+        master_metadata = get_remote_file(self.metadata_path)
+        # run_command(f"git show origin/master:{str(PurePosixPath(PureWindowsPath(self.metadata_path)))}")
```
Comment (amshamah419) [inline demisto_sdk/commands/update_release_notes/update_rn.py:142]: Remove

**Review by amshamah419 — APPROVED** [review]
Looks good to me, just make sure to update the change log.

File: demisto_sdk/__main__.py [hunk 4]
```diff
@@ -901,7 +902,13 @@ def update_pack_releasenotes(**kwargs):
         sys.exit(0)
     if _packs:
         for pack in _packs:
-            if pack in packs_existing_rn and update_type is not None:
+            if pack in packs_existing_rn and update_type is None:
+                try:
+                    with open(packs_existing_rn[pack], 'r') as f:
+                        prev_rn_text = f.read()
+                except Exception as e:
+                    print_error(f'Failed to load prev rn file content: {e}')
```
Comment (roysagi) [inline demisto_sdk/__main__.py:910]:
```suggestion
                    print_error(f'Failed to load the previous release notes file content: {e}')
```

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 5]
```diff
@@ -129,6 +133,19 @@ def _does_pack_metadata_exist(self):
 
         return True
 
+    def get_master_version(self):
+        """
+        Get the current version from origin/master if available, otherwise return '0.0.0'
+        """
+        master_current_version = '0.0.0'
+        master_metadata = get_remote_file(self.metadata_path)
```
Comment (roysagi) [inline demisto_sdk/commands/update_release_notes/update_rn.py:141]: what happens if the file does not exist on master - maybe we need try/except here?

Comment (orhovy) [reply demisto_sdk/commands/update_release_notes/update_rn.py:141]: You are right, I added an try exception, with an error saying that master isnt reachable so the version will be taken from the local metadata file.

**Pull request merged**